draft-ietf-quic-multipath.md

```diff
@@ -462,16 +524,43 @@ peer during the address validation procedure. According to
 {{QUIC-TRANSPORT}}, the standard way to deny the establishment of a path
 is to not send a PATH_RESPONSE in response to the peer's PATH_CHALLENGE.
 
-### Effect of RETIRE_CONNECTION_ID Frame {#retire-cid-close}
 
-Receiving a RETIRE_CONNECTION_ID frame causes an endpoint to discard
-the resources associated with that Connection ID. Specifically, the endpoint
-should not use the sequence number of the retired Connection ID anymore in
-any control frames, as the peer will not be able to associate those frames to
-a path and will therefore ignore them. This means an endpoint is also not required
-to acknowledge any late packets carrying that Connection ID and, hence,
-it can remove the list of received packets used to send acknowledgements after
-receiving the RETIRE_CONNECTION_ID frame.
+### Allocating, Consuming and Retiring Connection IDs {#consume-retire-cid}
+
+Each endpoints pre-allocate a Path Identifier for each new Connection ID. 
+The Path Identifier 0 indicates the initial path of the connection. 
+Endpoints SHOULD issue at least one unused Connection ID for each path.
+
+An endpoint maintains a set of connection IDs received from its peer for each path, 
+any of which it can use when sending packets, as the same in {{QUIC-TRANSPORT}}. 
+The difference of multi-path extension is that Connection IDs are pre-allocated
+for each paths. Each Connection ID is belonging to one path specified by
+the Path Identifier field of MP_NEW_CONNECTION_ID frame in {{mp-new-conn-id-frame}}.
+The Connection IDs used during handshake are belonging to the initial path
+with Path Identifier 0.
+
+When the endpoint wishes to remove a connection ID from use, it sends 
+a MP_RETIRE_CONNECTION_ID frame {{mp-retire-conn-id-frame}} to its peer. 
+Sending a MP_RETIRE_CONNECTION_ID frame indicates that the connection ID 
+will not be used again and requests that the peer replace it with a new connection ID 
+using a MP_NEW_CONNECTION_ID frame.
+
```

Comment:
I think adding an MP_RETIRE_CONNECTION_ID is unnecessary, and complicates the design. It is simpler to keep connection sequence number unique globally, and leave the RETIRE CONNECTION ID unchanged.

Reply:
I think that a per-path CID sequence number space is simpler, in particular since the `MP_NEW_CONNECTION_ID` has a `Retire Prior To` field. Retiring CIDs (which could be in use) for other paths when you get new CIDs for a path seems complex and not very useful.

Reply:
I created issue #296. Please continue the discussion there.